### DISCUSSION TITLE
Easiest way to import Zotero references from a LibreOffice (.odt) document into Quarto with VS Code Extension?

### DISCUSSION THREAD

**Discussion by jarisdata** [quarto-dev/quarto-cli#9577] [Q&A] [answered]
### Description

Hi everyone,
I would like to use Quarto to publish my PhD thesis and integrate graphs from Jupyter Notebooks, preferably using something similar to Tufte CSS: https://github.com/fredguth/tufte-quarto

Currently, all the text, including the Zotero references, is still in a LibreOffice document (.odt). For the coding part I work in VS Code. I saw there is a VS Code Extension for Quarto so I could potentially keep working within VS Code. 

1) What would be a useful workflow to make that transition, given the combination of tools I use?

2) Is there a (rather painless) way to import/copy the text from the odt document into Quarto without loosing the Zotero references.

Hints or links to guides would be much appreciated.

Best,
Jari

**Comment by cscheid** [comment]
> Is there a (rather painless) way to import/copy the text from the odt document into Quarto without loosing the Zotero references.

There's nothing painless like that, unfortunately. You could try to manually create a conversion script using Pandoc to markdown, but I'm certain that it will require additional work on your part. Quarto really does assume that you'll start from `.qmd` files or `.ipynb` files, and not from a separate document.

**Reply by jarisdata** [reply]
Thanks @cscheid, I wasn't really expecting a painless solution (fits neatly into the overall PhD experience).

 I will give conversion with Pandoc a try and post any relevant solution here. 

I guess the visual markdown editor that comes with the Quarto VS Code Extension manages the integration with Zotero. Are there any up- or downsides that come with working in a `.ipynb` or `.qmd` file for use with Zotero or [Juris-M](https://juris-m.readthedocs.io/en/latest/index.html) (Zotero client with support for non-Western naming conventions) in this setup?

Absolutely amazing project btw. Many thanks!!!

**Reply by cscheid** [reply]
> (fits neatly into the overall PhD experience)

I was a tenured professor before joining the Quarto project, so you've given me an excuse to say:

focus on finishing the PhD rather in whatever is the easiest way. Maybe stick with .odt only for now! :)

> Are there any up- or downsides that come with working in a `.ipynb` or `.qmd` file for use with Zotero or [Juris-M](https://juris-m.readthedocs.io/en/latest/index.html) (Zotero client with support for non-Western naming conventions) in this setup?

I unfortunately have no experience with Juris-M.

**Reply by jarisdata** [reply, marked as answer]
What would a PhD be without a little structured procrastination...

With these steps I was able to transfer my Zotero references from the LibreOffice .odt to a Quarto .qmd

1.  in the .odt file switch to ["Better Bibtex Citekeys"](https://retorque.re/zotero-better-bibtex/citing/migrating/better-bibtex-citekeys.csl)  citation style, see [here](https://retorque.re/zotero-better-bibtex/citing/migrating/)
2. save the .odt as .docx in LibreOffice 
3. in the .docx go to Zotero Document Peferences, choose "Switch Word Processor", and save transfer document, as described [here](https://www.zotero.org/support/kb/moving_documents_between_word_processorsl)
4. open the document in Microsoft Word and refresh references (Unfortunately, that requires MS Word to communicate with Zotero, so when you are on Linux you will have to run MS Word, Zotero, and LibreOffice from a Windows VM - Office in Bottles didn't do for me)
5. from the .docx extract the references as CSL JSON using this [reference extractor](https://rintze.zelle.me/ref-extractor/)
6. Open a terminal and covert the .docx to .qmd using Pandoc: quarto pandoc -f docx+citations -t markdown -i input.docx -o output.qmd
7. in your Quarto YAML file/header add the extracted references ("bibliography: ref-extracts.json")
